### PR TITLE
[#2209] Skip blank lines in blurb file url

### LINKED ISSUE #2209: blurbs.md: Allow blank lines before repo/branch
Current:

In the `blurbs.md`, the following works:
```
https://github.com/org1/repo1/tree/master

some info about repo1

<!--repo-->--------------------------------------------------------
https://github.com/org2/repo2/tree/master

some info about repo2
```
but the following does not (due to the blank line below the separator):
```
https://github.com/org1/repo1/tree/master

some info about repo1

<!--repo-->--------------------------------------------------------

https://github.com/org2/repo2/tree/master

some info about repo2
```

Suggestion: While not very important, we can also allow the 2nd one, simply by doing a `trim()/strip()` before processing each block of text.

Justification: the 2nd one is easier on the eye, when authoring the `blurbs.md`.

### PR DESCRIPTION
**Pull request by jedkohjk**
Resolves #2209

### Proposed commit message

```
Ignore blank lines after delimiter in blurbs file

Uses a loop to skip blank lines when searching for the relevant url.
```

### CODE REVIEW

File: src/main/java/reposense/parser/BlurbMarkdownParser.java [hunk 1]
```diff
@@ -120,9 +124,16 @@ private UrlRecord getUrlRecord(List<String> lines, int position) throws InvalidM
...
         try {
-            String url = lines.get(position).strip();
+            String url;
+            // skips blank lines
+            for (url = ""; url.length() == 0; url = lines.get(position++).strip()) {
+                // checks if delimiter is the last non-blank line
+                if (position >= lines.size()) {
+                    return null;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (jedkohjk) [reply src/main/java/reposense/parser/BlurbMarkdownParser.java]: Updated to use a while loop

**Comment by jedkohjk** [comment]
> Thanks for working on this @jedkohjk! Could you update the unit and system tests to include test cases for this scenario?

Modified blurbs files in test resources to account for this scenario

**Review by ckcherry23 — APPROVED** [review]
LGTM! Great work!

**Comment by ckcherry23** [comment]
@jedkohjk Can you update all your PR titles to include the issue number as per the expected format?

**Comment by jedkohjk** [comment]
> @jedkohjk Can you update all your PR titles to include the issue number as per the expected format?

Updated. Sorry, didn't notice the format previously.

**Review by gok99 — APPROVED** [review]
LGTM

**Pull request merged**